src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -43,50 +35,33 @@ protected override void OnBalanceChanging(ApplicationEngine engine, UInt160 acco
                 StorageItem storage_validator = engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_Candidate).Add(state.VoteTo));
                 CandidateState state_validator = storage_validator.GetInteroperable<CandidateState>();
                 state_validator.Votes += amount;
-            }
-        }
-
-        private void DistributeGas(ApplicationEngine engine, UInt160 account, NeoAccountState state)
-        {
-            BigInteger gas = CalculateBonus(state.Balance, state.BalanceHeight, engine.Snapshot.PersistingBlock.Index);
-            state.BalanceHeight = engine.Snapshot.PersistingBlock.Index;
-            GAS.Mint(engine, account, gas);
-        }
 
-        private BigInteger CalculateBonus(BigInteger value, uint start, uint end)
-        {
-            if (value.IsZero || start >= end) return BigInteger.Zero;
-            if (value.Sign < 0) throw new ArgumentOutOfRangeException(nameof(value));
-            BigInteger amount = BigInteger.Zero;
-            uint ustart = start / Blockchain.DecrementInterval;
-            if (ustart < Blockchain.GenerationAmount.Length)
-            {
-                uint istart = start % Blockchain.DecrementInterval;
-                uint uend = end / Blockchain.DecrementInterval;
-                uint iend = end % Blockchain.DecrementInterval;
-                if (uend >= Blockchain.GenerationAmount.Length)
-                {
-                    uend = (uint)Blockchain.GenerationAmount.Length;
-                    iend = 0;
-                }
-                if (iend == 0)
+                if (state_validator.Votes == 0)
```

Comment:
If the old value is `0` and the new value is not `0`, you should create entry for `Prefix_VoterRewardPerCommittee`.

Reply:
Only when a `VoteTo` becomes a committee?

Reply:
Currently, it was created at https://github.com/neo-project/neo/pull/1716/files#diff-fcd3074d4a84851394a505553ebcd177R134

Reply:
Currently, if a node get votes in block `N`, and lose it's votes in block `N+1`, we won't create record for it. Because it create record in `OnPersist()` of block `N+1`, and then be deleted by this `if` section.

Reply:
But before we delete `Prefix_VoterRewardPerCommittee` here, we have already called `DistributeGas`, which means all voters of this committee have already been rewarded.

Reply:
Where do you call `DistributeGas()`?

Reply:
Here, https://github.com/neo-project/neo/blob/master/src/neo/SmartContract/Native/Tokens/NeoToken.cs#L42

Reply:
What if the balance has not changed?

Reply:
If so, we'll not delete `Prefix_VoterRewardPerCommittee`.

Reply:
I can set `VoteTo` to `null` without transfer my neo. What do you do in this case if I am the last voter to the committee member?

Reply:
The incentives don't change, but `Prefix_VoterRewardPerCommittee`  will never be deleted. Fixed in https://github.com/neo-project/neo/pull/1716/commits/32db697244906bd733a47b1dccc0b6e8ecb85f64